# TRF-metaT

This is the code for the analysis of the microbial composition (16S), functional potential (functional metagenomics), and gene expression (metatranscriptomics) of mice exposed to high-fat diet (HFD) and treated with the intervetion time-restricted feeding (TRF). The goal was to characterize the microbial activity of microbes under HFD and TRF and use what we learn of microbial expression to guide the engineering of new bacterial therapeutics.

## Instructions on Data Access

### Multi-omics

We used samples collected and phenotyped from [Zarrinpar, et al. 2014](https://www.cell.com/cell-metabolism/fulltext/S1550-4131(14)00505-1). The 16S data was sequenced as part of this earlier study. Shotgun metagenomics and metatranscriptomics are available in Qiita [https://qiita.ucsd.edu] under the study ID 14501. Untargeted metabolomics data collected on the new engineered microbes is available in the MassIVE database [https://massive.ucsd.edu/] under the MassIVE ID MSV000094578. Feature based molecular networking data is available in [GNPS2](https://gnps2.org/status?task=f67a6015e87942fab913beedd63a36aa).

## Organization of files

### Input data
All data used for analysis is available under `./data`

### Data Analysis

## Citation

Flores Ramos, S., Siguenza, N., Zhong, W., Mohanty, I., Lingaraju, A., Richter, R.A., Karthikeyan, S., Lukowski, A.L., Zhu, Q., Nunes, W.D.G., Xu, Z.Z., Hasty, J., Dorrestein, P.C., Panda, S., Knight, R., Zarrinpar, A. “Metatranscriptomics Yields Bacterial Transgenes with Pronounced Metabolic Effects”. (submitted)
